### PR TITLE
Mind transfer for wizards

### LINKED ISSUE #20028: Being cloned and/or getting Staff-of-Change'd makes you lose all your spells
#### Description of issue
title
i feel like spells are tied to the body rather than the mind for some reason

#### Difference between expected and actual behavior
be crewman with book magic
get turned into a martian | get killed and cloned
retain spells
vs
be crewman with book magic
get turned into a martian | get killed and cloned
lose all spells

#### Length of time in which bug has been known to occur
at the latest since role datums, though no specific time

#### Server revision
d9cb1d08d578e7d63f29e8d681386e438000faf0

### PR DESCRIPTION
**Pull request by ShiftyRail**
Closes #20028 (for wiznerds)

### CODE REVIEW

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
https://github.com/vgstation-coders/vgstation13/issues/19873
https://github.com/vgstation-coders/vgstation13/issues/20029
https://github.com/vgstation-coders/vgstation13/issues/19964
This seems to be a problem with most spells in the game not just wizzie

**Comment by DamianX** [comment]
\>doesn't even compile
why bother PRing

**Comment by ShiftyRail** [comment]
I pressed push before checking and now I pay the price

**Comment by ShiftyRail** [comment]
image
This works (additionally, it compiles now)

**Comment by DamianX** [comment]
Should this be a thing?

**Comment by Shadowmech88** [comment]
Logic-wise, I don't see why magic spells in general shouldn't be tied to the mind. Every creature is evidently capable of learning spells from books, so that knowledge shouldn't be lost upon cloning.
Balance-wise, no idea.

File: code/datums/gamemode/role/role.dm [hunk 1]
```diff
@@ -571,6 +571,12 @@
 	to_chat(antag.current, "<span class='info'><a HREF='?src=\ref[antag.current];getwiki=[wikiroute]'>(Wiki Guide)</a></span>")
 
 
+/datum/role/wizard/handle_mind_transfer(var/mob/living/new_character, var/mob/living/old_character)
+	. = ..()
+	for (var/spell/S in old_character.spell_list)
+		if (S.user_type == USER_TYPE_WIZARD)
+			new_character.add_spell(S)
```
Comment (DamianX) [inline code/datums/gamemode/role/role.dm:578]: By looking at the implementation of `add_spell` it seems this wouldn't remove the spell from the old mind, is that the case?

**Comment by ShiftyRail** [comment]
It doesn't indeed. Just like in the old days, you can become a wizard too.
You just need to fix the body you destroyed.

**Comment by DamianX** [comment]
Well but this will let everyone get the wizard's spell rather than just one person. Personally I don't give a shit, just want to know it's intended.

**Comment by ShiftyRail** [comment]
I consider this a fair trade now that you need to fix (the possibly fucked up beyond repair) wizman body and you also need to get a wizard who accepts to be cloned & killed several times.

**Pull request merged**